### PR TITLE
Move speakers, sessions, schedule data to Firebase database #202

### PR DESCRIPTION
**Pull request by sophieH29**
Move data to Firebase https://github.com/gdg-x/hoverboard/issues/202
- speakers
- schedule
- sessions
- partners
- videos
- team

### CODE REVIEW

File: src/elements/app-data.html [hunk 1]
```diff
@@ -1,17 +1,99 @@
...
 <dom-module id="app-data">
   <template>
-    <iron-ajax auto url="[[speakersSource]]" handle-as="json" last-response="{{_speakersRaw}}" on-response="_updateSpeakers"></iron-ajax>
-    <iron-ajax auto url="[[sessionsSource]]" handle-as="json" last-response="{{sessions}}"></iron-ajax>
-    <iron-ajax auto url="[[postsSource]]" handle-as="json" on-response="_reversePosts"></iron-ajax>
-    <iron-ajax auto url="[[partnersSource]]" handle-as="json" last-response="{{app.partners}}"></iron-ajax>
-    <iron-ajax auto url="[[scheduleSource]]" handle-as="json" last-response="{{_scheduleRaw}}"></iron-ajax>
-    <iron-ajax auto url="[[teamSource]]" handle-as="json" last-response="{{app.team}}"></iron-ajax>
-    <iron-ajax auto url="[[videosSource]]" handle-as="json" last-response="{{app.videos}}"></iron-ajax>
+    <iron-ajax auto url="[[configSource]]" handle-as="json" last-response="{{app.data}}">
```
Comment (ozasadnyy) [inline src/elements/app-data.html]: We don't load `hoverboard.config` anymore. As a result, we don't have `app.data`

File: src/elements/app-data.html [hunk 2]
```diff
@@ -1,17 +1,99 @@
...
+    </iron-ajax> <iron-ajax auto url="[[postsSource]]" handle-as="json" on-response="_reversePosts"></iron-ajax>
+
+    <template is="dom-if" if="[[app.data]]">
```
Comment (ozasadnyy) [inline src/elements/app-data.html]: Remove `dom-if`

File: src/elements/app-data.html [hunk 3]
```diff
@@ -1,17 +1,99 @@
...
+    <firebase-document
+      id="querySpeakers"
+      app-name="[[app.data.firebase.name]]"
```
Comment (ozasadnyy) [inline src/elements/app-data.html]: Replace with `{$ firebase.name $}` and all occurrences

File: src/elements/app-data.html [hunk 4]
```diff
@@ -1,17 +1,99 @@
...
+      path="/speakers"
+      data="{{_firebaseSpeakersRaw}}">
+    </firebase-document>
+
+    <app-indexeddb-mirror
+      session="[[app.data.firebase.indexedDbSession]]"
```
Comment (ozasadnyy) [inline src/elements/app-data.html]: Replace with `{$ firebase.indexedDbSession$}` and all occurrences

File: src/elements/app-data.html [hunk 5]
```diff
@@ -31,41 +113,11 @@
             value: 'data/hoverboard.config.json'
```
Comment (ozasadnyy) [inline src/elements/app-data.html]: Please, remove `configSource` as well

File: src/elements/app-data.html [hunk 6]
```diff
@@ -84,14 +136,6 @@
           schedule: {
             type: Array,
             notify: true
-          },
-
-          _speakersRaw: {
```
Comment (ozasadnyy) [inline src/elements/app-data.html]: Why have you removed these properties if you use them in `_generateSchedule()` ?

File: src/elements/app-data.html [hunk 7]
```diff
@@ -130,7 +183,8 @@
         },
 
         _getSession: function (sessionId, day, dayIndex, schedule) {
-          var session = this.sessions[sessionId];
+          var session = this.sessions[sessionId] || this.sessions[sessionId.id];
```
Comment (ozasadnyy) [inline src/elements/app-data.html]: Can `sessionId` be as an object?

File: src/elements/app-data.html (hunk 6 above)
Comment (sophieH29) [inline src/elements/app-data.html]: I think it is redundant to initialize them, coz their initialized while retrieving from FB

 <app-indexeddb-mirror
      session="{$ firebase.indexedDbSession $}"
      key="speakers"
      data="{{_firebaseSpeakersRaw}}"
      persisted-data="{{_speakersRaw}}">
    </app-indexeddb-mirror>

**Pull request merged**